src/flash.rs

```diff
@@ -0,0 +1,246 @@
+use crate::stm32::FLASH;
+
+/// The first address of flash memory
+pub const FLASH_START: u32 = 0x0800_0000;
+pub const FLASH_END: u32 = 0x0801_FFFF;
+
+/// The size of a Flash memory page, in bytes
+pub const PAGE_SIZE: u32 = 2048;
+/// How many Flash memory pages there are
+pub const NUM_PAGES: u32 = 64;
```

Comment:
TODO: some devices have less memory.

Reply:
So we could have a feature to select the flash size, with largest option the default. And/Or have more specific device features (optionally). e.g. `--features stm32g071gb` 

Which are implemented as

```
stm32g071c8 = ["stm32g0/stm32g071", "stm32g0x1", "flash_64k"]
stm32g071gb = ["stm32g0/stm32g071", "stm32g0x1", "flash_128k"]
```

But we will need to define all possible options I guess

Reply:
I haven't yet looked how others have solved it.